pkg/auth/azure.go

```diff
@@ -278,34 +229,43 @@ func (a *Azure) getBearerTokenForApplication() (string, error) {
 	return tokenResponse.AccessToken, nil
 }
 
-func (a *Azure) getBearerTokenOnBehalfOfUser(token string) (string, error) {
-	form := url.Values{}
-	form.Add("client_id", a.clientID)
-	form.Add("client_secret", a.clientSecret)
-	form.Add("scope", "https://graph.microsoft.com/.default")
-	form.Add("grant_type", "urn:ietf:params:oauth:grant-type:jwt-bearer")
-	form.Add("requested_token_use", "on_behalf_of")
-	form.Add("assertion", token)
+func (a *Azure) GetGroupID(groupMail string) (string, error){
+	token, err := a.getBearerTokenForApplication()
+	if err != nil {
+		return "",err
+	}
 
-	req, err := http.NewRequest(http.MethodPost, endpoints.AzureAD(a.tenantID).TokenURL, strings.NewReader(form.Encode()))
+	params := url.Values{}
+	params.Add("$select", "id,displayName,mail")
+	params.Add("$filter", fmt.Sprintf("mail eq '%v'", groupMail))
+
+	req, err := http.NewRequest(http.MethodGet, 
+		AzureGroupsEndpoint+ "?"+params.Encode(),
+		nil)
 	if err != nil {
+		fmt.Print(err)
```

Comment:
Hvorfor skriver du ut feilen i denne funksjonen, holder det ikke å returnere den? Hvis vi trenger å skrive den ut her er det bedre å bruke loggrammeverket.

Reply:
Jeg glemt å fjerne, det var for debug:D